src/components/searchBar/index.tsx

```diff
@@ -39,21 +47,33 @@ export default function SearchBar() {
   );
 
   const handleDeleteIngredient = useCallback((index: number) => {
-    setSelectedIngredientsList((prevState) => prevState.filter((item) => item.id !== index));
+    setSelectedIngredientsList((prevState) =>
+      prevState.filter((item) => item.id !== index)
+    );
   }, []);
 
   return (
     <>
       <input
         className="flex w-full mt-4 px-6 py-4 rounded bg-gray-300 border border-gray-600 text-sm text-gray-900"
         type="text"
-        value={value}
+        value={searchQuery}
         onChange={handleChangeInput}
         placeholder="TYPE YOUR INGREDIENTS HERE"
       />
-      {searchResultList && (
+      {isLoading && (
+        <strong className="px-2 py-4 text-teal-600 hover:text-teal-900">
+          Loading
+        </strong>
+      )}
```

Comment:
Is react query compatible with react suspense? Can we take advantage of it and abstract our loading ui to a dedicated layer?

Reply:
React Query can be used with React's new Suspense for Data Fetching API's. To enable this mode, we can set either the global or query level config's suspense option to true.

I could replace the loading por something like this:

```suggestion
     <React.Suspense fallback={<h1>Loading ingredients...</h1>}>
              {showIngredients ? (
                  <Ingredients/>
                ) 
               : null}
            </React.Suspense>
```